rand.go

```diff
@@ -0,0 +1,47 @@
+package retry
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var r = &lockedSource{src: rand.New(rand.NewSource(time.Now().UnixNano()))}
+
+type lockedSource struct {
+	src *rand.Rand
+	lk  sync.Mutex
+}
+
+var _ rand.Source = (*lockedSource)(nil)
+
+// Int63 mimics math/rand.(*Rand).Int63 with mutex locked.
+func (r *lockedSource) Int63() int64 {
+	r.lk.Lock()
+	n := r.src.Int63()
```

Comment:
This can panic, meaning we'll never unlock. I think we should take the hit on the `defer` to ensure we release the lock.